opentracing-impl/src/test/java/io/opentracing/AbstractTracerTest.java

```diff
@@ -89,6 +89,23 @@ public void testNonEmptyExtract() {
         assertEquals("Should find the marker", "whatever", ((TestSpanBuilder)result).operationName);
     }
 
+    @Test
+    public void testExtractAsParent() throws Exception {
+        Map<String,String> map = Collections.singletonMap("test-marker", "whatever");
+        TextMapExtractAdapter adapter = new TextMapExtractAdapter(map);
+        AbstractTracer tracer = new TestTracerImpl();
+        SpanContext parent = tracer.extract(Format.Builtin.TEXT_MAP, adapter);
+        assert AbstractNoopSpan.INSTANCE != tracer.buildSpan("child").asChildOf(parent).start();
+    }
+
+    @Test
+    public void testExtractOfNoParent() throws Exception {
+        AbstractTracer tracer = new TestTracerImpl();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf((Span)AbstractNoopSpan.INSTANCE).start();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf((SpanContext)AbstractNoopSpan.INSTANCE).start();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf(AbstractNoopSpanBuilder.INSTANCE).start();
```

Comment:
> I think the most common behavior of the tracers is actually the opposite: continue the trace if there is one, start a new one otherwise.

This is a decision to be made (but I would like this decision not to interfere with this PR).

Currently there is no mechanism to indicate one way of the other.
In lei of that it makes most sense, and is the safest approach, not to intentionally initialise and start new traces. Otherwise this would cause an massive amplification of trace data.

This goes back to the original code snippet I posted that you would expect to write in library or framework code.

    public Response serverProcessingRequest(Requestr request) {
        TextMapExtractAdapter adapter = new TextMapExtractAdapter(request.getHeaders());
        SpanContext parent = tracer.extract(Format.Builtin.TEXT_MAP, adapter);
        try (Span span = tracer.buildSpan("some-name").asChildOf(parent).start()) {
            // server processes request
            return response;
        }
    }

> I intentionally haven't touched on the forcing tracing where extract returns a noop. That is functionality to add: and given it has not yet been dealt with; it's outside of the scope of this PR.

The instrumentation of libraries and frameworks **must not** dictate (or force) the sampling of traces. This is something that must be inherited either through the trace context (see Zipkin's `SAMPLED` header) or via a sampling mechanism used. In lei of that mechanism the default needs to be safe.